src/en/hstream/src/eu/kanade/tachiyomi/animeextension/en/hstream/Hstream.kt

```diff
@@ -0,0 +1,440 @@
+            anime.title = element.select("h2").text()
+            anime.thumbnail_url = baseUrl + element.select("img").attr("src")
+            return anime
+        }
+    }
+
+    override fun searchAnimeNextPageSelector(): String = "ul.pagination li a[rel=next]"
+
+    override fun searchAnimeSelector(): String = "article.bs"
+
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList): Request {
+        val parameters = getSearchParameters(filters)
+        return if (query.isNotEmpty()) {
+            filterSearch = false
+            GET("$baseUrl/search?s=${query.replace(("[\\W]").toRegex(), " ")}") // regular search
+        } else {
+            filterSearch = true
+            GET("$baseUrl/hentai/search?$parameters&page=$page") // filter search //work
+        }
+    }
+
+    // Details
+
+    override fun animeDetailsParse(document: Document): SAnime {
+        val anime = SAnime.create()
+        anime.thumbnail_url = baseUrl + document.selectFirst("div.thumbook div.thumb img")!!.attr("src")
+        val title = document.select("div.infox h1").text()
+        anime.title = title
+        anime.genre = document.select("div.info-content div.genxed a")
+            .joinToString(", ") { it.text() }
+        anime.description = document.select("div.info-content div.desc").text().substringAfter("Watch $title on HentaiStream.moe in 720p 1080p and (if available) 2160p (4k).").trim()
+        anime.author = document.select("div.info-content div.spe span:nth-child(2)  a")
+            .joinToString(", ") { it.text() }
+        anime.status = parseStatus(document.select("div.info-content div.spe span:nth-child(1)").text().trim().substringAfter("Status: "))
+        return anime
+    }
+
+    private fun parseStatus(statusString: String): Int {
+        return when (statusString) {
+            "Ongoing" -> SAnime.ONGOING
+            else -> SAnime.COMPLETED
+        }
+    }
+
+    // Latest
+
+    override fun latestUpdatesSelector(): String = "article.bs"
+
+    override fun latestUpdatesRequest(page: Int): Request = GET("$baseUrl/hentai/search?order=latest&page=$page")
+
+    override fun latestUpdatesFromElement(element: Element): SAnime {
+        val anime = SAnime.create()
+        anime.setUrlWithoutDomain(baseUrl + element.select("a").attr("href"))
+        anime.title = element.select("h2").text()
+        anime.thumbnail_url = baseUrl + element.select("img").attr("src")
+        return anime
+    }
+
+    override fun latestUpdatesNextPageSelector(): String = "ul.pagination li a[rel=next]"
+
+    // Settings
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        val videoQualityPref = ListPreference(screen.context).apply {
+            key = PREF_QUALITY_KEY
+            title = PREF_QUALITY_TITLE
+            entries = PREF_QUALITY_ENTRIES
+            entryValues = PREF_QUALITY_ENTRIES
+            setDefaultValue("720p")
```

Comment:
why 720p as default?

Reply:
Not many people pref 1080p or 4k all the time. They can always change it in settings

Reply:
Since the 4k videos are ups aled from 720p in most cases, I guess it's ok.